scripts/db-heal-test.py

```diff
@@ -0,0 +1,284 @@
+
+# Create working dir which gets deleted after the run
+working_dir = os.path.join(tmp_path, 'db-heal-test')
+if os.path.exists(working_dir):
+    shutil.rmtree(working_dir)
+os.makedirs(working_dir)
+
+# Log file path from which we read output to find kill_block
+aida_log_file = os.path.join(working_dir, 'aida.log')
+carmen_log_file = os.path.join(working_dir, 'carmen.log')
+genesis = os.path.join(working_dir, 'test_genesis.dat')
+
+current_dir = Path.cwd()
+carmen_root = os.path.join(current_dir, 'go')
+if "scripts" in str(current_dir):
+    carmen_root = os.path.abspath('../go')
+
+print("Your settings:")
+print(f"\tNumber of iterations: {number_of_iterations}.")
+print(f"\tSync time before kill: {window} seconds.")
+print(f"\tCheckpoint granularity: {checkpoint_granularity} blocks.")
+
+
+# Function which checks programs return code, if program failed, log is printed and True is returned.
+def has_program_failed(code, log):
+    if code != 0:
+        log.close()
+        with open(carmen_log_file, 'r') as l:
+            text = l.read()
+            print(text)
+        return True
+    return False
+
+
+# Function which checks every line added to aida_log_file and behaves accordingly to the line.
+def check_aida_log(sleep_time: int, checkpoint: int):
+    start = 0.0
+    with open(aida_log_file, 'r') as f:
+        while True:
+            line = f.readline()
+            if not line:
+                time.sleep(0.1)
+                continue
+            if start > 0 and time.time() - start >= sleep_time:
+                print("Interrupting...")
+                process.terminate()
+                return False
+            if start == 0.0 and "Track: block" in line:
+                # This line in combination with 0 start time means Syncing has been restarted.
+                print("Syncing restarted!")
+                start = time.time()
+            # If process ends with error (return code 1) or either 'fail' or 'exit status' occurs in line exit script
+            if process.poll() == 1 or any(s in line for s in ["exit status", "fail"]):
+                print("Error occurred - printing output.log:")
+                with open(aida_log_file, 'r') as l:
+                    text = l.read()
+                    print(text)
+                return True
+
+
+# Function which runs Carmen's info command and finds the latest checkpoint from created log
+def get_latest_checkpoint_from_info():
+    log = os.path.join(working_dir, 'carmen-info.log')
+    cp: str
+    with open(log, 'w') as cl:
+        r = subprocess.run(
+            ['go', 'run', './database/mpt/tool', 'info', str(archive)],
+            stdout=cl,
+            stderr=cl,
+            cwd=carmen_root)
+        if has_program_failed(r.returncode, cl):
+            return -1
+
+    with open(log, 'r') as cl:
+        info_checkpoint = cl.readlines()[-1]
+        # Return last word which is the block number
+        cp = info_checkpoint.split()[-1]
+
+    os.remove(log)
```

Comment:
I guess you do not have to remove the log here, right? It will get removed together with the whole `tmp` dir